relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -192,24 +283,37 @@ impl SamplingRule {
 /// The fields will be used by rules to check if they apply.
 trait FieldValueProvider {
     /// gets the value of a field
-    fn get_value(&self, path: &str) -> FieldValue;
+    fn get_value(&self, path: &str) -> Value;
     /// what type of rule can be applied to this provider
     fn get_rule_type(&self) -> RuleType;
+    /// returns a filtering function for custom operators.
+    /// The function returned takes the provider and a condition definition and
+    /// returns a match result
+    fn get_custom_operator(
+        name: &str,
+    ) -> fn(condition: &CustomCondition, slf: &Self, ip_addr: Option<IpAddr>) -> bool;
```

Comment:
Q: Why not just run the custom condition inside the function? 

```rust
fn match_custom(&self, condition: &CustomCondition, ip_addr: Option<IpAddr>) -> bool;
```

For a follow-up: Later down the road we need to refactor the value provider so that it can expose all fields needed by the custom rules via its `get_value` interface.

Reply:
We could do that, it seemed wrong though. 
The ValueProvider gives you information from the Event or Trace but doesn't also run 
rules, anyway not feeling too strongly either way.
The presence of the custom operation in ValueProvider makes it quite a messy abstraction.

Reply:
Separate from this PR, could you compile a list of all attributes that we need access to for running the custom matchers?